### ISSUE TITLE
[v1.1.1] controller manager fails immediately when api-server is not up 

### ISSUE THREAD

**Issue by dalanlan** [kubernetes/kubernetes#17109]
IIUC, `controller manager` is expected to wait 10s before it fails when `api-server` is not up (Ref #16413). Yet it doesn't.

```
$ sudo vi /var/log/upstart/kube-controller-manager.log 
I1111 19:25:03.490354    2309 plugins.go:71] No cloud provider specified.
I1111 19:25:03.490428    2309 nodecontroller.go:133] Sending events to api server.
E1111 19:25:03.490625    2309 nodecontroller.go:203] Error monitoring node status: Get http://127.0.0.1:8080/api/v1/nodes: dial tcp 127.0.0.1:8080: connection refused
E1111 19:25:03.490684    2309 controllermanager.go:290] Failed to start service controller: ServiceController should not be run without a cloudprovider.
F1111 19:25:03.490795    2309 controllermanager.go:314] Failed to get api versions from server: Get http://127.0.0.1:8080/api: dial tcp 127.0.0.1:8080: connection refused
~
```
k8s version

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"1", GitVersion:"v1.1.1", GitCommit:"92635e23dfafb2ddc828c8ac6c03c7a7205a84d8", GitTreeState:"clean"}
Server Version: version.Info{Major:"1", Minor:"1", GitVersion:"v1.1.1", GitCommit:"92635e23dfafb2ddc828c8ac6c03c7a7205a84d8", GitTreeState:"clean"}
```

**Comment by MikeSpreitzer** [comment]
Should the warning about this issue be removed from https://github.com/kubernetes/kubernetes/blob/master/docs/getting-started-guides/ubuntu.md#configure-and-start-the-kubernetes-cluster ?

**Comment by dalanlan** [comment]
@MikeSpreitzer  Nope, it is fixed in v1.1.2, still suffer in v1.1.1 to my best knowledge

**Comment by MikeSpreitzer** [comment]
v1.1.2 is the latest, right?
If the problem is fixed in HEAD, then the warning about it can be removed from the doc in HEAD, right?

**Comment by MikeSpreitzer** [comment]
Oh, I see your point.  The warning is specifically about version 1.1.1, and the problem remains in that version.  If that is accurate then the warning should remain (although an added remark that the problem is fixed in 1.1.2 would not hurt).

**Comment by dalanlan** [comment]
I had a commit for that in #17616 , yet was missing by mistake:-/ 
I'll add it back then.

**Comment by nikhiljindal** [comment]
Agreed that docs on HEAD should reflect the state on HEAD.
If someone wants to look at 1.1 docs, they should do so at http://kubernetes.io/v1.1/docs/

**Comment by nikhiljindal** [comment]
oh I see now that the warning says that the issue is only in 1.1.1 and is fixed in 1.1.2.
That LGTM.
Thx!

I guess we can close this issue now?

**Comment by MikeSpreitzer** [comment]
*this* issue was already closed.  I generated some confusion under another issue, number 17666, because I read the web instead of HEAD. I have closed that issue too.
